### PR TITLE
[ONNX] Onnx node tests

### PR DESCRIPTION
**Pull request by mbrookhart**
We've been hitting a lot of errors running models with ONNX, that has led to a lot of piecewise fixes. This is an attempt to fix the importer more broadly by running the tests onnx ships with pip https://github.com/onnx/onnx/tree/master/onnx/backend/test/data/node

These files contain an onnx graph, input arrays, and expected outputs, so we can test directly against the canonical onnx tests. This PR provides a method to import these tests as parameterized unit tests, execute them, and skip any we know currently fail. I also fixed a lot of low hanging fruit to reduce the number of skipped unit tests.

Future PRs will work to fix the currently skipped tests, and then extend this to GPU.

For reference, this is the pytest result on my system, testing against ONNX 1.6, which is what we have in CI:

`434 passed, 123 skipped, 83 deselected, 1185 warnings in 32.40s`

This adds a lot of tests, but they are all small, so the runtime is actually pretty minuscule, and it improves our ONNX import coverage dramatically.

cc @jwfromm @masahi @jroesch @electriclilies @adelbertc

### CODE REVIEW

File: tests/python/frontend/onnx/test_forward.py [hunk 1]
```diff
@@ -4090,6 +4090,170 @@ def verify_cumsum(indata, axis, exclusive=0, reverse=0, type="float32"):
...
+            new_tensor = onnx.TensorProto()
+            with open(tensor, "rb") as f:
+                new_tensor.ParseFromString(f.read())
+            if "input" in tensor.split("/")[-1]:
+                inputs.append(numpy_helper.to_array(new_tensor))
+            elif "output" in tensor.split("/")[-1]:
+                outputs.append(numpy_helper.to_array(new_tensor))
+            else:
+                print(tensor)
+                raise
```
Comment (electriclilies) [inline tests/python/frontend/onnx/test_forward.py:4248]: Can you put an error message here? Maybe something like "Expected tensor to be either an input or output"

**Review by electriclilies — COMMENTED** [review]
Overall looks good to me, just a few nitpicks

File: python/tvm/relay/frontend/onnx.py [hunk 2]
```diff
@@ -444,9 +448,15 @@ class ConvTranspose(OnnxOpConverter):
     @classmethod
     def _impl_v1(cls, inputs, attr, params):
         # get number of channels
-        channels = infer_channels(inputs[1], True)
+        out_type = infer_type(inputs[1])
+        out_shapes = [get_const_tuple(out_type.checked_type.shape)]
+        channels = out_shapes[0][1]
```
[the remark this answers is not in the document]
Comment (mbrookhart) [reply python/tvm/relay/frontend/onnx.py:452]: ONNX always assumes NCHW

[the remark this answers is not in the document]
Comment (electriclilies) [reply python/tvm/relay/frontend/onnx.py:452]: Cool, just wanted to make sure we didn't have to worry about it!

**Review by electriclilies — APPROVED** [review]

**Review by jroesch — APPROVED** [review]
In the next PR can you just paste you rational above the test section so that new comers understand what's going on?

**Pull request merged**